src/plugins/irc-events/error.js

```diff
@@ -2,21 +2,57 @@ var Msg = require("../../models/msg");
 
 module.exports = function(irc, network) {
 	var client = this;
-	irc.on("errors", function(data) {
+
+	// TODO: remove later
+	irc.on("irc_error", function(data) {
+		console.log("Got an irc_error");
+		irc.emit("error", data);
+	});
+
+	irc.on("error", function(data) {
+		console.log("error", data);
+		var text = data.error;
+		if (data.reason) {
+			text = data.reason + " (" + text + ")";
+		}
 		var lobby = network.channels[0];
 		var msg = new Msg({
 			type: Msg.Type.ERROR,
-			text: data.message,
+			text: text,
 		});
 		client.emit("msg", {
 			chan: lobby.id,
 			msg: msg
 		});
-		if (!network.connected) {
-			if (data.cmd === "ERR_NICKNAMEINUSE") {
-				var random = irc.me + Math.floor(10 + (Math.random() * 89));
-				irc.nick(random);
-			}
-		}
+	});
+
+	irc.on("nick in use", function(data) {
+		var lobby = network.channels[0];
+		var msg = new Msg({
+			type: Msg.Type.ERROR,
+			text: "Nickname " + data.nick + " is already in use: " + data.reason,
```

Comment:
We could either:
- Not care about which `nick` caused it and let the network decide if they want to or not.
- Prepend or append something:
  - `text: "Nick change to " + data.nick + "attempted: ` + data.reason` (doesn't look good if the reason is empty, if it's even possible)
  - `text: (data.reason + " (" + data.nick + ")") || ("Nickname " + data.nick + " is already in use.")` or `text: (data.reason || "Nickname is already in use.") + " (" + data.nick + ")"`
- (Since I'm listing options, I'm allowed for some overkill :D) Search for `data.nick` in `data.reason` and append the nickame is not present

That's all I can think of right now. Maybe not ideal but I think better than having twice the reason on Freenode for example.
Also, this is more intriguing, I think:

> A bug I noticed though: if the nickname is already taken, /nick fails but the user still gets assigned a new random nickname, instead of not changing nickname.